src/wallet/rpcwallet.cpp

```diff
@@ -2397,6 +2399,40 @@ UniValue lockunspent(const JSONRPCRequest& request)
 
         COutPoint outpt(uint256S(txid), nOutput);
 
+        const bool isLocked = pwallet->IsLockedCoin(outpt.hash, outpt.n);
+
+        if (fUnlock)
```

Comment:
Thanks for the review! Are you absolutely sure that attempting to lock an already locked output should be an error?